tests/unit/test_dns.cpp

```diff
@@ -145,6 +145,31 @@ void TestDnsName(void)
 
     message->SetOffset(0);
 
+    printf("----------------------------------------------------------------\n");
+    printf("Verify domain name match:\n");
+
+    {
+        const char *subDomain;
+        const char *domain;
+
+        subDomain = "my-service._ipps._tcp.local.";
+        domain    = "local.";
+        VerifyOrQuit(Dns::Name::IsSubDomainOf(subDomain, domain), "Name::IsSubDomainOf() failed");
```

Comment:
```suggestion
        subDomain = "my-service._ipps._tcp.local.";
        domain    = "local.";
        VerifyOrQuit(Dns::Name::IsSubDomainOf(subDomain, domain), "Name::IsSubDomainOf() failed");
        
        subDomain = "my-service._ipps._tcp.local";
        domain    = "local.";
        VerifyOrQuit(Dns::Name::IsSubDomainOf(subDomain, domain), "Name::IsSubDomainOf() failed");
        
        subDomain = "my-service._ipps._tcp.local.";
        domain    = "local";
        VerifyOrQuit(Dns::Name::IsSubDomainOf(subDomain, domain), "Name::IsSubDomainOf() failed");

        subDomain = "my-service._ipps._tcp.local";
        domain    = "local";
        VerifyOrQuit(Dns::Name::IsSubDomainOf(subDomain, domain), "Name::IsSubDomainOf() failed");
```

Test different combination of names that end with dot or not?

Reply:
updated.